src/data_structures/bit_tree.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2014-2015 Johannes Köster, Vadim Nazarov, Patrick Marks
```

Comment:
Has Vadim contributed to this? I think it should be you and your colleagues only, Patrick. Not even me until I'll ever touch this code.